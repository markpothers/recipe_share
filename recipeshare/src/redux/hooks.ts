import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import type { AppDispatch } from "./store"
import type { RootStateType } from "./types"

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootStateType> = useSelector
