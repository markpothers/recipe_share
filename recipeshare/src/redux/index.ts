export { useAppDispatch, useAppSelector } from "./hooks";
export * from "./types";
export type { RootState } from "./store"
export {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn
} from "./rootReducer"
export { default as rootReducer } from "./rootReducer"
