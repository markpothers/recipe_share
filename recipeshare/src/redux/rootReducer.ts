import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { ListChef, ListRecipe, LoginChef, MakePic, MakePicChef, Recipe, Comment } from "../centralTypes";
import { initialRootState } from "./initialRootState";
import {
	ParameterContent,
	RootStateType,
	StoreChefDetails,
	StoreNewFollowers,
	UpdateSingleChefList,
	UpdateSingleRecipeList,
} from "./types";

const rootReducer = createSlice({
	name: "root",
	initialState: initialRootState,
	reducers: {
		updateSingleRecipeList(state, action: PayloadAction<UpdateSingleRecipeList>) {
			state.allRecipeLists[action.payload.listKey] = action.payload.recipeList;
		},
		updateAllRecipeLists(state, action: PayloadAction<Record<string, ListRecipe[]>>) {
			state.allRecipeLists = action.payload;
		},
		updateSingleChefList(state, action: PayloadAction<UpdateSingleChefList>) {
			state.allChefLists[action.payload.listKey] = action.payload.chefList;
		},
		updateAllChefLists(state, action: PayloadAction<Record<string, ListChef[]>>) {
			state.allChefLists = action.payload;
		},
		storeRecipeDetails(state, action: PayloadAction<Recipe>) {
			state.recipe_details = action.payload;
		},
		updateNewUserDetails(state, action: PayloadAction<ParameterContent>) {
			state.newUserDetails[action.payload.parameter] = action.payload.content;
		},
		clearNewUserDetails(state) {
			state.newUserDetails = initialRootState.newUserDetails;
		},
		updateLoginUserDetails(state, action: PayloadAction<ParameterContent>) {
			state.loginUserDetails[action.payload.parameter] = action.payload.content;
		},
		clearLoginUserDetails(state) {
			state.loginUserDetails = initialRootState.loginUserDetails;
		},
		addRecipeLike(state) {
			state.recipe_details.recipe_likes += 1;
			state.recipe_details.likeable = false;
		},
		removeRecipeLike(state) {
			state.recipe_details.recipe_likes -= 1;
			state.recipe_details.likeable = true;
		},
		addRecipeReShare(state) {
			state.recipe_details.re_shares += 1;
			state.recipe_details.shareable = false;
		},
		removeRecipeReShare(state) {
			state.recipe_details.re_shares -= 1;
			state.recipe_details.shareable = true;
		},
		addRecipeMake(state) {
			state.recipe_details.recipe_makes += 1;
			state.recipe_details.makeable = false;
		},
		addMakePic(state, action: PayloadAction<MakePic>) {
			state.recipe_details.make_pics.unshift(action.payload);
		},
		addMakePicChef(state, action: PayloadAction<MakePicChef>) {
			state.recipe_details.make_pics_chefs.unshift(action.payload);
		},
		saveRemainingMakePics(state, action: PayloadAction<MakePic[]>) {
			state.recipe_details.make_pics = action.payload;
		},
		updateComments(state, action: PayloadAction<Comment[]>) {
			state.recipe_details.comments = action.payload;
		},
		storeNewFollowers(state, action: PayloadAction<StoreNewFollowers>) {
			state.chef_details[action.payload.chefID].followers = action.payload.followers;
			state.chef_details[action.payload.chefID].chef_followed =
				!state.chef_details[action.payload.chefID].chef_followed;
		},
		updateLoggedInChef(state, action: PayloadAction<RootStateType["loggedInChef"]>) {
			state.loggedInChef = {
				id: action.payload.id,
				e_mail: action.payload.e_mail,
				username: action.payload.username,
				auth_token: action.payload.auth_token,
				image_url: action.payload.image_url,
				is_admin: action.payload.is_admin,
				is_member: action.payload.is_member,
			};
		},
		storeChefDetails(state, action: PayloadAction<StoreChefDetails>) {
			state.chef_details[action.payload.chefID] = action.payload.chef_details;
		},
		clearChefDetails(state) {
			state.chef_details = initialRootState.chef_details;
		},
		stayLoggedIn(state, action: PayloadAction<boolean>) {
			state.stayLoggedIn = action.payload;
		},
	},
});

// dispatch(stayLoggedIn(true))

export const {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn,
} = rootReducer.actions;

export default rootReducer.reducer;
