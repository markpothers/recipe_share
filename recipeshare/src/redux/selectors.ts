import { ReduxStateType } from "./types";
import { createSelector } from "@reduxjs/toolkit";

const selectRoot = (state: ReduxStateType) => state.root

export const getLoggedInChef = createSelector(selectRoot, root => root.loggedInChef)
export const getLoggedInUserDetails = createSelector(selectRoot, root => root.loginUserDetails)
export const getDeviceType = createSelector(selectRoot, root => root.deviceType)
export const getStayLoggedIn = createSelector(selectRoot, root => root.stayLoggedIn)
