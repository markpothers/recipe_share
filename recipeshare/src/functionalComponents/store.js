import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

const initialState = {
  loggedInChef: {
    id: "",
    username: "",
    auth_token: "",
    imageURL: "",
    is_admin: false
  },
  recipes: {
    all: [],
    chef: [],
    chef_feed: [],
    chef_liked: [],
    chef_made: [],
    global_ranks: [],
    most_liked: [],
    most_made: []
  },
  recipes_details: {

  },
  newRecipeDetails: {
    name: "",
    instructions: ['test', 'test2'],
    ingredients: {
      ingredient1 :{
        name:"",
        quantity: "",
        unit: "Oz"
      }
    },
    difficulty: "0",
    time: "00:15",
    imageBase64: "",
    filter_settings: {
      "Breakfast": false,
      "Lunch": false,
      "Dinner": false,
      "Chicken": false,
      "Red meat": false,
      "Seafood": false,
      "Vegetarian": false,
      "Salad": false,
      "Vegan": false,
      "Soup": false,
      "Dessert": false,
      "Side": false,
      "Whole 30": false,
      "Paleo": false,
      "Freezer meal": false,
      "Keto": false,
      "Weeknight": false,
      "Weekend": false,
      "Gluten free": false,
      "Bread": false,
      "Dairy free": false,
      "White meat": false,
    },
    cuisine: "Any",
    serves: "Any",
    acknowledgement: ""
  },
  newUserDetails: {
    first_name: "",
    last_name: "",
    username: "",
    e_mail: "",
    password: "",
    password_confirmation: "",
    country: "United States",
    imageURL: "",
    profile_text: ""
  },
  loginUserDetails: {
    e_mail: "markpothers@hotmail.com",
    password: "123456"
  },
  chefs: {
    all_chefs: [],
    followed: [],
    most_liked_chefs: [],
    most_made_chefs: [],
    chef_followees: [],
    chef_followers: []
  },
  chefs_details: {},
  filter_settings: {
    "Breakfast": false,
    "Lunch": false,
    "Dinner": false,
    "Chicken": false,
    "Red meat": false,
    "Seafood": false,
    "Vegetarian": false,
    "Salad": false,
    "Vegan": false,
    "Soup": false,
    "Dessert": false,
    "Side": false,
    "Whole 30": false,
    "Paleo": false,
    "Freezer meal": false,
    "Keto": false,
    "Weeknight": false,
    "Weekend": false,
    "Gluten free": false,
    "Bread": false,
    "Dairy free": false,
    "White meat": false,
  },
  cuisine: "Any",
  serves: "Any",
  stayLoggedIn: false,
}

  const middleware = compose(
    applyMiddleware(ReduxThunk)
  )

  export const store = createStore(reducer, initialState, middleware)
