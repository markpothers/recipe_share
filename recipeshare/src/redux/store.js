import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

// import { filters } from '../dataComponents/filters'

//exported ONLY for unit testing
export const initialState = {
	loggedInChef: {
		id: "",
		username: "",
		auth_token: "",
		image_url: "",
		is_admin: false
	},
	allRecipeLists: {},
	allChefLists: {},
	recipes_details: {

	},
	newRecipeDetails: {
		name: "",
		instructions: [
			'',
			'',
			'',
			'',
		],
		instructionImages: [],
		ingredients: {
			ingredient1: {
				name: "",
				quantity: "",
				unit: "Oz"
			}
		},
		difficulty: "0",
		time: "00:15",
		imageBase64: "",
		filter_settings: {
			"Breakfast": false,
			"Lunch": false,
			"Dinner": false,
			"Chicken": false,
			"Red meat": false,
			"Seafood": false,
			"Vegetarian": false,
			"Salad": false,
			"Vegan": false,
			"Soup": false,
			"Dessert": false,
			"Side": false,
			"Whole 30": false,
			"Paleo": false,
			"Freezer meal": false,
			"Keto": false,
			"Weeknight": false,
			"Weekend": false,
			"Gluten free": false,
			"Bread": false,
			"Dairy free": false,
			"White meat": false,
		},
		cuisine: "Any",
		serves: "Any",
		acknowledgement: ""
	},
	newUserDetails: {
		first_name: "",
		last_name: "",
		username: "",
		e_mail: "",
		password: "",
		password_confirmation: "",
		country: "United States",
		image_url: "",
		profile_text: ""
	},
	loginUserDetails: {
		e_mail: "",
		password: ""
	},
	chefs: {
		all_chefs: [],
		followed: [],
		most_liked_chefs: [],
		most_made_chefs: [],
		chef_followees: [],
		chef_followers: []
	},
	chefs_details: {},
	filter_settings: {
		"Breakfast": false,
		"Lunch": false,
		"Dinner": false,
		"Chicken": false,
		"Red meat": false,
		"Seafood": false,
		"Vegetarian": false,
		"Salad": false,
		"Vegan": false,
		"Soup": false,
		"Dessert": false,
		"Side": false,
		"Whole 30": false,
		"Paleo": false,
		"Freezer meal": false,
		"Keto": false,
		"Weeknight": false,
		"Weekend": false,
		"Gluten free": false,
		"Bread": false,
		"Dairy free": false,
		"White meat": false,
	},
	filterCuisines: {
		all: "Any",
		chef: "Any",
		chef_feed: "Any",
		chef_liked: "Any",
		chef_made: "Any",
		global_ranks: "Any",
		most_liked: "Any",
		most_made: "Any"
	},
	serves: "Any",
	stayLoggedIn: false,
}

export const middleware = compose( //exported ONLY for unit testing
	applyMiddleware(ReduxThunk)
)

export const store = createStore(reducer, initialState, middleware)
