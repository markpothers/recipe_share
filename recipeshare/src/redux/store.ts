import { createStore, compose, applyMiddleware } from "redux";
import ReduxThunk from "redux-thunk";
import reducer from "./reducer";
import { RootState } from "./types";

//exported ONLY for unit testing
export const initialState: RootState = {
	loggedInChef: {
		id: "",
		username: "",
		auth_token: "",
		image_url: "",
		is_admin: false,
	},
	allRecipeLists: {},
	allChefLists: {},
	recipe_details: {
		chef_id: 0,
		chef_username: "",
		comments: [],
		ingredient_uses: [],
		ingredients: [],
		instructions: [],
		instruction_images: [],
		likeable: true,
		make_pics: [],
		make_pics_chefs: [],
		makeable: true,
		re_shares: 0,
		recipe: {
			acknowledgement: null,
			acknowledgement_link: null,
			chef_id: 0,
			cook_time: 0,
			created_at: "",
			cuisine: "Any",
			description: "",
			difficulty: 0,
			hidden: false,
			id: 0,
			name: "",
			prep_time: 0,
			serves: "Any",
			show_blog_preview: false,
			time: 0,
			total_time: 0,
			updated_at: "",
			Breakfast: false,
			Lunch: false,
			Dinner: false,
			Chicken: false,
			"Red meat": false,
			Seafood: false,
			Vegetarian: false,
			Salad: false,
			Vegan: false,
			Soup: false,
			Dessert: false,
			Side: false,
			"Whole 30": false,
			Paleo: false,
			"Freezer meal": false,
			Keto: false,
			Weeknight: false,
			Weekend: false,
			"Gluten free": false,
			Bread: false,
			"Dairy free": false,
			"White meat": false,
		},
		recipe_images: [],
		recipe_likes: 0,
		recipe_makes: 0,
		shareable: true,
	},
	newRecipeDetails: {
		name: "",
		instructions: ["", "", "", ""],
		instructionImages: [],
		ingredients: {
			ingredient1: {
				name: "",
				quantity: "",
				unit: "Oz",
			},
		},
		difficulty: "0",
		time: "00:15",
		imageBase64: "",
		filter_settings: {
			Breakfast: false,
			Lunch: false,
			Dinner: false,
			Chicken: false,
			"Red meat": false,
			Seafood: false,
			Vegetarian: false,
			Salad: false,
			Vegan: false,
			Soup: false,
			Dessert: false,
			Side: false,
			"Whole 30": false,
			Paleo: false,
			"Freezer meal": false,
			Keto: false,
			Weeknight: false,
			Weekend: false,
			"Gluten free": false,
			Bread: false,
			"Dairy free": false,
			"White meat": false,
		},
		cuisine: "Any",
		serves: "Any",
		acknowledgement: "",
	},
	newUserDetails: {
		first_name: "",
		last_name: "",
		username: "",
		e_mail: "",
		password: "",
		password_confirmation: "",
		country: "United States",
		image_url: "",
		profile_text: "",
	},
	loginUserDetails: {
		e_mail: "",
		password: "",
	},
	chef_details: {
		chef: {
			country: "United States",
			created_at: "",
			id: 0,
			image_url: null,
			profile_text: "",
			username: "",
		},
		chef_commented: false,
		chef_followed: false,
		chef_liked: false,
		chef_made: false,
		chef_make_piced: false,
		chef_shared: false,
		comments: 0,
		comments_received: 0,
		followers: 0,
		following: 0,
		make_pics: 0,
		make_pics_received: 0,
		re_shares: 0,
		re_shares_received: 0,
		recipe_likes: 0,
		recipe_likes_received: 0,
		recipes: 0,
	},
	// filter_settings: {
	// 	Breakfast: false,
	// 	Lunch: false,
	// 	Dinner: false,
	// 	Chicken: false,
	// 	"Red meat": false,
	// 	Seafood: false,
	// 	Vegetarian: false,
	// 	Salad: false,
	// 	Vegan: false,
	// 	Soup: false,
	// 	Dessert: false,
	// 	Side: false,
	// 	"Whole 30": false,
	// 	Paleo: false,
	// 	"Freezer meal": false,
	// 	Keto: false,
	// 	Weeknight: false,
	// 	Weekend: false,
	// 	"Gluten free": false,
	// 	Bread: false,
	// 	"Dairy free": false,
	// 	"White meat": false,
	// },
	// filterCuisines: {
	// 	all: "Any",
	// 	chef: "Any",
	// 	chef_feed: "Any",
	// 	chef_liked: "Any",
	// 	chef_made: "Any",
	// 	global_ranks: "Any",
	// 	most_liked: "Any",
	// 	most_made: "Any"
	// },
	// serves: "Any",
	stayLoggedIn: false,
};

export const middleware = compose(
	//exported ONLY for unit testing
	applyMiddleware(ReduxThunk)
);

export const store = createStore(reducer, initialState, middleware);

export type AppDispatch = typeof store.dispatch;
