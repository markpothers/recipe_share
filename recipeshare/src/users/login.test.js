import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux'
import { store } from '../redux/store'

import LoginScreen from './login.js'
import { TouchableOpacity, TextInput } from 'react-native';

describe('Login', () => {

	let component

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(() => {
		// console.log('runs before every test')
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	test('renders; has 2 TextInputs and 5 TouchableOpacities', () => {
		act(() => {
			component = renderer.create(
				<Provider
					store={store}
				>
					<LoginScreen/>
				</Provider>
			)
		})
		const image = component.toJSON()
		expect(image).toMatchSnapshot()
		let root = component.root
		const inputs = root.findAllByType(TextInput)
		expect(inputs.length).toEqual(2)
		const buttons = root.findAllByType(TouchableOpacity)
		expect(buttons.length).toEqual(7)
	})

});
