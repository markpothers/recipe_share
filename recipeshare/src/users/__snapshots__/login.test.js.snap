// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders; has 2 TextInputs and 5 TouchableOpacities 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "borderWidth": 0,
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 0,
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      nestedScrollEnabled={true}
      scrollEnabled={true}
      style={
        Object {
          "borderColor": "yellow",
          "borderWidth": 0,
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "top": 133.4,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 266.8,
                  "width": "75%",
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderWidth": 0,
                  "justifyContent": "center",
                  "marginLeft": 75,
                  "marginRight": 75,
                  "width": 600,
                },
                Object {
                  "marginTop": 200.1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 11.25,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.5}
                    style={
                      Object {
                        "borderRadius": 11.25,
                        "color": "#505050",
                        "fontSize": 33.35,
                        "fontWeight": "bold",
                        "height": "100%",
                        "overflow": "hidden",
                        "paddingHorizontal": 15,
                        "paddingVertical": 7.5,
                        "textAlign": "center",
                        "width": "100%",
                      }
                    }
                  >
                    Welcome, chef!
                    

                     Please log in or register
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 11.25,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="email"
                    keyboardType="email-address"
                    maxFontSizeMultiplier={2}
                    onChange={[Function]}
                    placeholder="e-mail"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    textContentType="username"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 11.25,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="password"
                    maxFontSizeMultiplier={2}
                    onChange={[Function]}
                    placeholder="password"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    textContentType="password"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "width": 112.5,
                        "zIndex": 0,
                      }
                    }
                  >
                    <Icon
                      name="eye"
                      size={53.36}
                      style={
                        Object {
                          "color": "#50505055",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 11.25,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": "50%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.25}
                    style={
                      Object {
                        "color": "#104e01",
                        "fontSize": 27.53633962602873,
                        "fontWeight": "bold",
                        "paddingHorizontal": 7.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Remember
                    

                    email
                  </Text>
                  <RCTSwitch
                    accessibilityRole="switch"
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    onTintColor="#4b714299"
                    style={
                      Object {
                        "height": 31,
                        "width": 51,
                      }
                    }
                    thumbTintColor={null}
                    value={false}
                  />
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": "50%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.5}
                    style={
                      Object {
                        "color": "#104e01",
                        "fontSize": 27.53633962602873,
                        "fontWeight": "bold",
                        "paddingHorizontal": 7.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Stay
                    

                     logged in
                  </Text>
                  <RCTSwitch
                    accessibilityRole="switch"
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    onTintColor="#4b714299"
                    style={
                      Object {
                        "height": 31,
                        "width": 51,
                      }
                    }
                    thumbTintColor={null}
                    value={false}
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 11.25,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": "50%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                    }
                  }
                >
                  <Icon
                    name="account-plus"
                    size={53.36}
                    style={
                      Object {
                        "color": "#104e01",
                      }
                    }
                  />
                  <Text
                    maxFontSizeMultiplier={1.5}
                    style={
                      Object {
                        "color": "#104e01",
                        "fontSize": 27.53633962602873,
                        "fontWeight": "bold",
                        "paddingHorizontal": 7.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Register
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": "50%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                    }
                  }
                >
                  <Icon
                    name="lock-open"
                    size={53.36}
                    style={
                      Object {
                        "color": "#104e01",
                      }
                    }
                  />
                  <Text
                    maxFontSizeMultiplier={1.7}
                    style={
                      Object {
                        "color": "#104e01",
                        "fontSize": 27.53633962602873,
                        "fontWeight": "bold",
                        "paddingHorizontal": 7.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Forgot
                    

                    password
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 11.25,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 11.25,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "center",
                      "maxWidth": "100%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                      "width": "100%",
                    }
                  }
                >
                  <Icon
                    name="login"
                    size={53.36}
                    style={
                      Object {
                        "color": "#104e01",
                      }
                    }
                  />
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#104e01",
                          "fontSize": 27.53633962602873,
                          "fontWeight": "bold",
                          "paddingHorizontal": 7.5,
                          "textAlign": "center",
                        },
                        Object {
                          "marginLeft": 30,
                        },
                      ]
                    }
                  >
                    Login
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
